dali/operators/math/normalize/normalize.cc

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/math/normalize/normalize.h"
+#include "dali/kernels/reduce/reduce.h"
+#include "dali/kernels/normalize/normalize_cpu.h"
+#include "dali/core/math_util.h"
+#include "dali/operators/math/normalize/normalize_utils.h"
+
+namespace dali {
+
+DALI_SCHEMA(Normalize)
+  .DocStr(R"(Normalizes the input by removing mean and dividing by standard deviation.
+
+The mean and standard deviation can be calculated internally for specified subset of axes or
+can be externally provided as `mean` and `stddev` arguments.
+
+The normalization is done following the formula::
+
+    out = scale * (in - mean) / stddev + shift
+
+The expression assumes that *out*, *in* are equally shaped tensors whereas *mean* and *stddev*
+may be either tensors of same shape or scalars or a mix of these. The expression follows
+*numpy* broadcasting rules.
+
+Sizes of (non-scalar) `mean` and `stddev` must have either extent of 1, if given axis is reduced,
+or match the corresponding extent of the input. A dimension is considered reduced if it's listed
+in `axes` or `axis_names`. If neither `axes` nor `axis_names` argument is
+present, the set of reduced axes is inferred by comparing input shape to the shape of
+mean/stddev arguments, but it is enforced that the set of reduced axes is the same for all tensors
+in the batch.
```

Comment:
A thought: A bunch of arg examples will help with understanding those paragraphs